high_card.rb

```diff
@@ -1,5 +1,117 @@
 # Use these two arrays to generate a deck of cards.
-ranks = ["A", 2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K" ]
-suits = [ "hearts", "spades", "clubs", "diamonds" ]
 
-players = []
+$winning_card = nil
+$second_winning_card = nil
+$players = []
+
+$r_ordered = [2,3,4,5,6,7,8,9,10,"J","Q","K","A"]
+# 0 to 12
+$s_ordered = ["clubs", "diamonds", "hearts", "spades"]
+# 0 to 3
+
+def shuffled_deck
+
+  new_deck = []
+  $r_ordered.each_with_index do |rank, index|
+    $s_ordered.each_with_index do |suit, index_two|
+      new_deck.push({
+        rank_n: index,
+        rank: rank,
+        suit_n: index_two,
+        suit: suit
+        })
+      end
+    end
+
+    shuffled = new_deck.shuffle
+    return shuffled
+  end
+
+def collect_players # fix this
+  number_of_players = 0
+  input = ""
+  player_array = []
+  while input != "play"
+    puts "#{number_of_players} so far. Enter a player name, or type 'play'."
```

Comment:
player_array.length      # number of players

You're right that runtime will be better without needless variables!

-Nathan

On Sat, Oct 31, 2015 at 4:32 PM, Nicholas Olds <notifications@github.com>
wrote:

> In high_card.rb
> <https://github.com/ga-dc/high_card/pull/75#discussion_r43575878>:
>
> > +        suit_n: index_two,
> > +        suit: suit
> > +        })
> > +      end
> > +    end
> > +
> > +    shuffled = new_deck.shuffle
> > +    return shuffled
> > +  end
> > +
> > +def collect_players # fix this
> > +  number_of_players = 0
> > +  input = ""
> > +  player_array = []
> > +  while input != "play"
> > +    puts "#{number_of_players} so far. Enter a player name, or type 'play'."
>
> Instead of number_of_players as a separate counter, can you think of a
> method on an your player_array that would the return the number of
> players?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ga-dc/high_card/pull/75/files#r43575878>.
>